### PR TITLE
GOV/DOC: add section to docs on triaging and triage team

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

This adds documentation to add a "triage team" to our organization as discussed on today's call.  The purpose of this is to:

 - provide a low barrier to formally recognize users as contributing to the maintenance of Matplotlib  (by giving them power and responsibility)
 - empower more people to help where they are willing and able
 - to incentive more people to triage issues
 - (secondary) provide a ramp / pipeline to contributing PRs (and reviewing PRs!) for new contributors

This text is based on https://scikit-learn.org/dev/developers/bug_triaging.html  and modified to better fit Matplotlib.

Task that need to be done immediately  outside of a PR if we merge this:

 - create triage_team in the github ACL system (I think only me @dopplershift and @efiring can currently do that?) with triage rights on matplotlib/matplotlib
 
If this catches on we may want to:

 - add more formal method of requesting to join (ex a form), but for now while we figure out if this is even a good idea, lets keep it informal.
 -  add "triage team captain" or similar to the named roles in the governance docs, but again lets see if this catches on / is a good idea before we do that.

On the call I suggested 1 triage would be the threshold to be invited, but ended up with 2-3 just so that we have a bit more context to evaluate.  Being optomistic, I expect most of these requests / nominations to be fulfilled.  

sklearn has a slightly higher bar for inviting people to the triage team (https://scikit-learn.org/dev/governance.html#triage-team) and more ceremony, but I would rather have this be broader and less formal.  

We can deal with bad-faith actors if they come up and I expect a vast majority of the issues we will have will be miss-labeled or miss-milestoned issues which are almost trivially fixable.

### CODE REVIEW

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:71]: please link here to the list of people w/ commit rights (which separately, a lot of folk with commit rights aren't active)  & possibly the default person they should reach out to (which is why my bias is towards a form) 'cause I can imagine folk not wanting to bother someone w/ commit rights - if I was in this boat when I started contributing, I would have asked Ryan 'cause I already knew him. Possible add another sentence here is "for example, one of the contributors that you have interacted with on an issue" to make it clear the contributors are on board w/ it & who they should be asking.

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:9]: I am totally anti the dev/user split, especially since like we don't even have these breakdowns in the issue trackers. Folks with commit rights open tons of bugs and feature requests, discussions about priorities tend to happen amongst whoever is commenting in that issue. suggested instead:
> The `issue tracker <https://github.com/matplotlib/matplotlib/issues>`_ is important to communication in the project because it serves as the centralized location for making feature requests, reporting bugs, identifying major projects to work on, and discussing priorities.

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:21]: kinda unclear what step means here - do you mean the tasks listed above? (milestone, label, close?)

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:37]: link to code format style here?

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:137]: this is the template - might be worth linking to the templates here - did they fill out the template- especially as these are variable depending on issue type.

**Review by story645 — CHANGES_REQUESTED** [review]

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:56]:
```suggestion
   different understanding of the process than a seasoned maintainer.
```

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:81]:
```suggestion
Triaging operations for members of the core and triage teams
------------------------------------------------------------
```

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:95]: And/or attempt a bisect?

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:130]:
```suggestion
   Matplotlib project itself, beyond just using the library. As such,
```

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:148]:
```suggestion
   if they can provide an example or simplify the provided one.
```

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:156]:
```suggestion
   OS, Python, and Matplotlib versions.
```

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:165]: Attempt to bisect here (if reproducer is simple, known good version exists, etc.)?

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:186]: We don't have an Enhancement label any more.
```suggestion
   about documentation, "Bug" if it is clearly a bug, "New feature" if it
   is a new feature request, ...
```

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:194]:
```suggestion
   the "GUI/Qt" label when relevant.
```

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:199]:
```suggestion
       `the scikit-learn project
```

File: doc/devel/triage.rst
Comment (tacaswell) [reply doc/devel/triage.rst:56]: This was un-edited from the sklean text, going to open a PR with them too :-p

File: doc/devel/triage.rst
Comment (tacaswell) [reply doc/devel/triage.rst:9]: I agree with the dev/user split being bad in general.  This particular wording didn't bother me too much as different people in the community do have different responsibilities, but have taken your wording here.

File: doc/devel/triage.rst
Comment (tacaswell) [reply doc/devel/triage.rst:37]: I think this is aimed at improving mark-down usage not the Python, clarified.

File: doc/devel/triage.rst
Comment (tacaswell) [reply doc/devel/triage.rst:176]: Added paragraph below (so GH did not notice).

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:178]:
```suggestion
   <https://git-scm.com/docs/git-bisect>`_ to find the first commit
```

File: doc/devel/triage.rst
Comment (QuLogic) [inline doc/devel/triage.rst:103]:
```suggestion
    determine the last working version.  Regressions should be
    milestoned for the next bug-fix release and may be labeled as
```

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:12]:
```suggestion
to issues and closing issues that are resolved or unresolvable
```re

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:73]:
```suggestion
2. Ask someone on the `triage team
```

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:92]:
```suggestion
 `available github labels
```

File: doc/devel/triage.rst
Comment (story645) [inline doc/devel/triage.rst:124]:
```suggestion
    question or has been considered as unclear for many years, then it
```

**Review by story645 — APPROVED** [review]

**Comment by jklymak** [comment]
@dopplershift @timhoffm @efiring   This is on the agenda for today, but if you guys are OK with adding a "triage" layer to the Membership then you can hit approve.  Thanks!

File: doc/devel/triage.rst
Comment (efiring) [inline doc/devel/triage.rst:77]: you -> your

**Review by dopplershift — APPROVED** [review]

**Review by efiring — APPROVED** [review]
Just one typo.
Looks good to me.

**Comment by tacaswell** [comment]
I will merge this when docs pass CI.

**Comment by tacaswell** [comment]
Testing @matplotlib/triageteam notification.

**Pull request merged**
